client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -509,6 +519,19 @@ protected String getPath(Method method, Object[] arguments, JaxRsMeta meta) {
                     } else {
                         path = path.replaceAll("\\{" + ((PathParam)annotation).value() + "\\}", this.urlEncode(this.serialize(value)));
                     }
+                } else if (annotation instanceof BeanParam) {
+                    if (value == null) {
+                        continue;
+                    }
+                    beanParamCache
+                        .computeIfAbsent(types[i], this::getBeanParamGetters)
+                        .forEach(v -> {
+                            Function<Object, Object> beanPropertyGetter = v.getValue1();
+                            Object beanPropertyValue = beanPropertyGetter.apply(value);
+                            Annotation[] annotations = v.getValue2();
+                            args.add(beanPropertyValue);
+                            argumentAnnotations.add(annotations);
```

Comment:
This changing of the list we are looping, both args and argumentAnnotations makes this REALLY hard to follow

Reply:
Why is it hard to follow? I think it is quite simple. While processing one argument, some new arguments are added.

As I have said before, I plan to rewrite the logic in HttpClient to make it functional and to avoid all reflection at runtime, but I would like to do that in a separate PR.